### ISSUE TITLE
Show 1st week of next month in calendar

### ISSUE THREAD

**Issue by gerazov** [nvim-orgmode/orgmode#98]
### Does this feature exist in Emacs orgmode core?

N/A

### Orgmode link

I've not used emacs.

### Feature value

It's easier to schedule tasks if one week of next month is also visible in the calendar view. Currently if I want to schedule something on Oct 1, and today is Sep 30, I have to press 7 keys: `>kkkkww`. With the modification 2 keys `ww`.

Actually, if we are optimizing the calendar, it would be nice to get rid of the pipes `|`, without them jumping between dates would be twice as fast :+1:

**Comment by kristijanhusak** [comment]
I agree it's easier, but it's not so easy to implement it. It would require much more logic, and a proper way to distinguish what belongs to next/prev month (highlighting and such).
I think simplest thing would be to jump to the 1st day of month when switching months, that would remove a need for additional movements.
Pipes are there because they delimit the day. If you would put a cursor in between two days (without the pipes) and tried to submit it, Vim would collect one or another, potentially something that you do not expect. I could add an additional validation to not allow submitting if cursor is not on a number though. I'm open for suggestions on that.

**Comment by gerazov** [comment]
Both make sense :+1:

**Comment by kristijanhusak** [comment]
I pushed a change that does 2 things:
1. It jumps to first day of month when going forward, or last day of month when going backward
2. Removed pipes and added a requirement for a cursor to be on a number

This makes using calendar a bit smoother. Let me know what you think.

**Comment by gerazov** [comment]
Awesome! :sunglasses: thanks :pray:

**Comment by gerazov** [comment]
I don't want to necrobump but I was about to create an issue about the same thing around one year later :sweat_smile:

**Comment by jgollenz** [comment]
I'm unsure I understand the issue fully. Do you propose that 
- the week that the last of the month is displayed in full, even if those days are not part of the current month
 
OR
- do you want a whole week of the next month displayed, regardless of which day of the week is the last of the current?

**Comment by gerazov** [comment]
The first one. It would be useful in weeks like the current one where Monday is 31st. But it's only a marginal thing if it's too difficult to implement :wink:

**Comment by jgollenz** [comment]
Reopening for now, else this will slip through the cracks.
